internal/btf/core.go

```diff
@@ -579,17 +647,39 @@ func coreFindField(local Type, localAcc coreAccessor, target Type) (_, _ coreFie
 				return coreField{}, coreField{}, err
 			}
 
-			if targetMember.BitfieldSize > 0 {
-				return coreField{}, coreField{}, fmt.Errorf("field %q is a bitfield: %w", targetMember.Name, ErrNotSupported)
-			}
-
 			local = localMember.Type
 			localMaybeFlex = acc == len(localMembers)-1
 			localOffset += localMember.OffsetBits
 			target = targetMember.Type
 			targetMaybeFlex = last
 			targetOffset += targetMember.OffsetBits
 
+			if targetMember.BitfieldSize > 0 {
+				// For bitfields we compute the offset from which to load
+				// the value, and we include the offset of the bitfield and its
+				// size in 'coreField' so we can later compute the proper bit shift.
+				size, offset, err := calculateBitfieldLoad(int(targetOffset), int(targetMember.BitfieldSize))
+				if err != nil {
+					return coreField{}, coreField{}, err
+				}
+
+				// Adjust the load instruction, if needed.
+				if targetLoad, ok := target.(*Int); ok && targetLoad.Size != uint32(size) {
+					target = target.copy()
+					target.(*Int).Size = uint32(size)
+				}
+
+				return coreField{local, localOffset, localMember.OffsetBits, localMember.BitfieldSize},
+					coreField{target, uint32(offset * 8), targetMember.OffsetBits, targetMember.BitfieldSize},
+					nil
+			} else if localMember.BitfieldSize > 0 {
```

Comment:
This is needed for `sk_type`, which transitions from `u32 sk_type:16` to `u16 sk_type`:
v5.16.12: https://elixir.bootlin.com/linux/v5.16.12/source/include/net/sock.h#L480
v4.20.17: https://elixir.bootlin.com/linux/v4.20.17/source/include/net/sock.h#L462

I think it should work, for this it would do a 16-bit load and then 0-sized shifts.